### PR TITLE
NOT INTENDED FOR MERGE: add basic layered deployment testing and testing templates

### PR DESCRIPTION
**Pull request by smcd253**
Added the following:
* basic layered deployment testing templates
* basic layered deployment template tests
* TODO:
## 4) Test invalid cases for all current and planned tests

NOTE: This is intended for preliminary code review only. Not intended for merge to master.

### CODE REVIEW

File: tests/assets/layered_deployment.template_2.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                                "createOptions": {}
+                            }
+                        },
+                        "LayeredDeploymentCSharpModule2": {
+                            "version": "1.0",
+                            "type": "docker",
+                            "status": "running",
+                            "restartPolicy": "always",
+                            "settings": {
+                                "image": "${MODULES.csharpmodule2}",
```
Comment (mhshami01) [inline tests/assets/layered_deployment.template_2.json:33]: should this be 'MODULES.csharpmodule2.amd64'?

Comment (smcd253) [reply tests/assets/layered_deployment.template_2.json:33]: I'm referring to deployment.template_3.json in the same directory, so I'm keeping it consistent with that until I start running the tests and tweak things around.

File: tests/assets/layered_deployment.template_1.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
+    "content": {
+        "modulesContent": {
+            "$edgeAgent": {
+                "properties.desired": {
+                    "modules": {
+                        "LayeredDeploymentTempSensor": {
+                            "settings": {
+                            "image": "mcr.microsoft.com/azureiotedge-simulated-temperature-sensor:${LAYERED_RUNTIME_TAG}",
```
Comment (cindydeng1998) [inline tests/assets/layered_deployment.template_1.json:9]: do we own this image? Or do we plan on updating it for layered deployment? I think the simulated temperature sensor image only goes to 1.0 for now

Comment (smcd253) [reply tests/assets/layered_deployment.template_1.json:9]: we do not. I copied the RUNTIME_TAG from the layered deployment tests and chose it to be separate for more control. but it is currently set to 1.0. I will change this to RUNTIME_TAG to keep consistency. this image is just for testing, so I can see the value in moving 1.0 to be included in this string but that's out of scope for this work.

**Comment by EliiseS** [comment]
@marianan @cindydeng1998 Should this PR be closed now then? This has been implemented in #498

**Comment by marianan** [comment]
Closing PR as functionality is implemented in separate PRs

**Pull request closed without merging**